sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/train_esm2.py

```diff
@@ -163,10 +164,17 @@ def main(
     strategy = nl.MegatronStrategy(
         tensor_model_parallel_size=tensor_model_parallel_size,
         pipeline_model_parallel_size=pipeline_model_parallel_size,
-        ddp="megatron",
+        pipeline_dtype=get_autocast_dtype(precision),
+        ddp=DistributedDataParallelConfig(
+            check_for_nan_in_grad=True,
+            overlap_grad_reduce=True,
+            overlap_param_gather=False,  # TODO waiting for a NeMo fix
+            average_in_collective=True,
+            use_distributed_optimizer=True,
```

Comment:
maybe make grad reduce in fp32 an option or set to True? True seems to be the default in nearly all of the nemo recipes. Also note that I'm not clear on whether the ddp grad reduce in fp32 or the megatron mixed precision takes precedent.

Reply:
The problem is `grad_reduce_fp32` hurts perf.

Reply:
Added in cli but disabled by default.